### PR TITLE
Github Workflow to generate binaries

### PR DESCRIPTION
**Pull request by rahuliyer95**
Simple GitHub workflow to generate Windows / macOS binaries. It will run with each push and create a binary against the master branch. Hope this helps to create releases :)

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Rust
+
+on: [push]
```
Comment (luisholanda) [inline .github/workflows/rust.yml]: Wouldn't it be better to only run in pushes to the `master` branch?

**Comment by luisholanda** [comment]
Is there a reason to not build Linux binaries?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (Kethku) [reply .github/workflows/rust.yml]: Actually it's useful for any branch I think. For situations when building might fail on a platform you don't have access to

Comment (luisholanda) [reply .github/workflows/rust.yml]: Yeah, makes sense 👍

**Comment by justinmk** [comment]
> Is there a reason to not build Linux binaries

Would need to be an appimage, else it is unlikely to work on most systems except Travis version of Ubuntu.

**Comment by rahuliyer95** [comment]
I'm not very familiar with building on Linux and the instructions in the README were not very clear to me. If you can provide me steps for building **appimage**  I can add it to the workflow. 

I was reading online and found out that Cargo can build native app bundles (like **neovide.app** on macOs) using [cargo-bundle](https://github.com/burtonageo/cargo-bundle) library. If you can set that up (or help get that working) we can have the workflow directly upload the app bundles as artifacts. 

Let me know what you think about it.

**Comment by justinmk** [comment]
> If you can provide me steps for building the **appimages** I can add it to the workflow

I can't provide steps, nor would I want you to spend your time on it :)

It requires reading appimage docs and takes some effort. Neovim's is [here](https://github.com/neovim/neovim/blob/master/scripts/genappimage.sh) for reference.

Linux has better dev tooling that mac/win should building neovide shouldn't be a problem. Though an appimage is great to have once someone puts the effort into it.

**Comment by Kethku** [comment]
This is a beautiful first start, so I'm going to merge it and split out the remaining tasks: build full packages using cargo-bundle, and add support for linux builds.

This is fantastic help. Thanks so much @rahuliyer95

**Pull request merged**